src/RateLimitedNotification.php

```diff
@@ -18,15 +21,16 @@ trait RateLimitedNotification
      */
     public function rateLimitKey($notification, $notifiables)
     {
-//        echo '<pre>';
-//        var_dump($notification);
-//        exit;
+
+        // Convert notifiables to Collection or Array
+        $notifiables = $this->formatNotifiables($notifiables);
+        $notifiablesIds = (is_array($notifiables)) ? md5(json_encode($notifiables)) : $notifiables->implode('id', ',');
```

Comment:
As I was reviewing the code in this, it appears that the way the cache key is built for multiple notifiables will not take into account the cache key for individual notifiables. Imagine a scenario where we have users 1 and 2, and only want to send  notification xyz every hour. Dispatching xyz to user 1 generates a cache key of 1, and then dispatching xyz again 1 second later, this time to users 1 and 2, generating a cache key of 1,2. Because the cache key does not match for user 1, they will still receive the notification twice, despite the rate limit

Reply:
There are two potential approaches to this.  My sense is that it might be best to provide both implementations and give users a choice of which they use, since each has tradeoffs and the "best" option may be different depending on your needs.

1. Each recipient gets their own cache key rather than building a combined cache key. This requires only minor changes, but with a large number of recipients, it also creates a significant number of individual cache keys which may not be the most efficient use of cache space. 
2. We stop relying on the cache to track expiries, and individually track the time that each user "last" received that particular notification (in an array stored in the cache).  

For now I'll assume that option 1 is 'sufficient' unless someone thinks that will really cause them a problem and we need to explore option 2 immediately?